cli/tests/test_split_command.rs

```diff
@@ -675,3 +680,199 @@ fn test_split_interactive_with_paths() {
     ◆  zzzzzzzz root() 00000000
     ");
 }
+
+// When a commit is split, the second commit produced by the split becomes the
+// working copy commit for all workspaces whose working copy commit was the
+// target of the split. This test does a split where the target commit is the
+// working copy commit for two different workspaces.
+#[test]
+fn test_split_with_multiple_workspaces_same_working_copy() {
+    let mut test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["git", "init", "main"]);
+    let main_path = test_env.env_root().join("main");
+    let secondary_path = test_env.env_root().join("secondary");
+
+    test_env.jj_cmd_ok(&main_path, &["desc", "-m", "first-commit"]);
+    std::fs::write(main_path.join("file1"), "foo").unwrap();
+    std::fs::write(main_path.join("file2"), "foo").unwrap();
+
+    // Create the second workspace and change its working copy commit to match
+    // the default workspace.
+    test_env.jj_cmd_ok(
+        &main_path,
+        &["workspace", "add", "--name", "second", "../secondary"],
+    );
+    // Change the working copy in the second workspace.
+    test_env.jj_cmd_ok(
+        &secondary_path,
+        &["edit", "-r", "description(first-commit)"],
+    );
+    let stdout = test_env.jj_cmd_success(&main_path, &["workspace", "list"]);
+    insta::assert_snapshot!(stdout, @r###"
+    default: qpvuntsm d12cb1c0 first-commit
+    second: qpvuntsm d12cb1c0 first-commit
+    "###);
+    // Check the working-copy commit in each workspace in the log output. The "@"
+    // node in the graph indicates the current workspace's working-copy commit.
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &main_path), @r###"
+    @  qpvuntsmwlqt default@ second@ first-commit
+    ◆  zzzzzzzzzzzz
+    "###);
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &secondary_path), @r###"
+    @  qpvuntsmwlqt default@ second@ first-commit
+    ◆  zzzzzzzzzzzz
+    "###);
```

Comment:
nit: This seems unnecessary. It should be tested in `test_workspaces.rs` (I didn't check, but I assume it is already tested there).

Reply:
Agreed. I think we can delete at least one of these log calls. I think it is nice to show the log output at least once though since it makes it easier to reason about how the graph looks after the split.

Reply:
Yes, I also like to have one for testing the setup and making it easier to follow. If you keep the log output here, I think you can drop the `jj workspace list` output above instead.

Reply:
Yep, I slimmed down the tests based on your feedback. Let me know if I misunderstood anything.